portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -76,97 +70,88 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
         long timestamp = (long) entity.getProperty("time");
         String tag = (String) entity.getProperty("tag");
         String comment = (String) entity.getProperty("text");
-        String nickname = (String) entity.getProperty("nickname");
+        String nickname = (String) entity.getProperty("sender");
         String email = (String) userService.getCurrentUser().getEmail();
         //String image = (String) entity.getProperty("imgUrl");
-        ArrayList<String> messageReplies = (ArrayList) entity.getProperty("replies");
-        //BlogMessage message = new BlogMessage(messageId, tag, comment, image, nickname, email, messageReplies, timestamp);
-        BlogMessage message = new BlogMessage(messageId, tag, comment, image, nickname, messageReplies, timestamp);
+        ArrayList<String> message_Replies = (ArrayList) entity.getProperty("replies");
+        BlogMessage message = new BlogMessage(messageId, tag, comment, nickname, email, message_Replies, timestamp);
         messages.add(message);
       }
       
       // Create BlogHashMap Object and put BlogMessages in the map.
       BlogHashMap blogMap = new BlogHashMap();
       blogMap.putInMap(messages);
 
-      // Load messages from BlogHashMap and respond with gson.
-      List<String> tagsToSearch = new ArrayList<String>();
-      tagsToSearch.add(""); // we'll get inputs later.
-
+       // Load messages from BlogHashMap and respond with gson.
       LinkedList<BlogMessage> loadedBlogMessages = blogMap.getMessages(tagsToSearch, numberOfCommentsToDisplay);
     
       Gson gson = new Gson();
       response.setContentType("application/json;");
       
       response.getWriter().println(gson.toJson(loadedBlogMessages));
       return;
-
     }
     
-
-    /**
-   * Converts a ServerStats instance into a JSON string using the Gson library
-   */
-    @Override
+   @Override
     public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+      // Get Post parameters.
       String message = request.getParameter("text-input");
-
-      String sender = getParameter(request, "sender", "Steven");
+      String sender = request.getParameter("sender");
+      String loadFactor = request.getParameter("load_factor");
+      String postTag = getTagParameter(request, "tags", InternalTags.defaultTag());
       
-      // TODO: 
-      //      Get default tag from the InternalTags class and use that below.
+      //TODO: Handle image file sent with FormData.
+      //String imageUrl = getUploadedFileUrl(request, imagePath);
       
-      // Get type of comment.
-      String commentType = getParameter(request, "tags", "Default");
-
-      //String imageUrl = getUploadedFileUrl(request, "image");
+      // Only put BlogMessages with a message and tag in datastore.
+      if (message.equals("") || message.equals(null) || postTag == null) {
```

Comment:
For, message.equals(""), use message.isEmpty()

For message.equals(null), I believe it always returns false.